lib/remote/apilistener.cpp

```diff
@@ -1575,16 +1589,14 @@ void ApiListener::ReplayLog(const JsonRpcConnection::Ptr& client)
 		}
 
 		if (last_sync) {
-			{
-				ObjectLock olock2(endpoint);
-				endpoint->SetSyncing(false);
-			}
-
 			OpenLogFile();
 
 			break;
 		}
-	}
+	} while (!stopReplay);
+
+	ObjectLock olock2 (endpoint);
+	endpoint->SetSyncing(false);
```

Comment:
Please consider using our `Defer` class for the cleanup work. This should make the code easier to read and more robust as you can simply use `return` instead of having to correctly exit the nested loops.